frontend/lib/Models/members_list_provider.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:adpv_frontend/DataModels/group_data.dart';
+import 'package:adpv_frontend/DataModels/member_summary.dart';
+import 'package:flutter/material.dart';
+
+enum SortingField { id, mail }
+
+enum SortingOrder { asc, desc }
+
+class Sorting {
+  SortingField field = SortingField.id;
+  SortingOrder order = SortingOrder.asc;
+}
+
+class MembersListProvider with ChangeNotifier {
+  List<MemberInfo> membersList = [];
+  int groupId;
+  Color emailColor = Colors.black;
+  Color idColor = Colors.pink;
+  Sorting currentSorting = Sorting();
+
+  MembersListProvider(this.groupId);
+
+  void makeMemberList(GroupData groupData) {
+    for (var e in groupData.members) {
+      membersList.add(MemberInfo(e.id, e.email, e.userRoles, e.otherGroups));
+    }
+    sortByIDAsc();
+    notifyListeners();
+  }
+
+  void delete(int id) {
+    membersList.removeWhere((element) => element.id == id);
+    notifyListeners();
+  }
+
+  void sortByIDAsc() {
+    membersList.sort((a, b) => a.id.compareTo(b.id));
+    notifyListeners();
+  }
+
+  void sortByIDDesc() {
+    membersList.sort((b, a) => a.id.compareTo(b.id));
+    notifyListeners();
+  }
+
+  void sortByEmailAsc() {
+    membersList.sort((a, b) => a.email.compareTo(b.email));
+    notifyListeners();
+  }
+
+  void sortByEmailDesc() {
+    membersList.sort((b, a) => a.email.compareTo(b.email));
+    notifyListeners();
+  }
```

Comment:
To mogą być metody prywatne + `notifyListeners` chyba nie jest tutaj potrzebne, bo i tak wywołujesz ją w metodzie `changeSorting`